lib/api/auth_user.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:http/http.dart' as http;
+
+
+class Auth {
+  final baseUrl = 'http://161.35.99.225/api/v1/auth/login';
+   String accessToken = '';
+   var id=0;
+  static String username = '';
+  static String password = '';
+  String basicAuth =
+      'Basic ${base64.encode(utf8.encode('$username:$password'))}';
+
+  String getUsername()  {
+    return username;
+  }
+
+  static void setUsername(String theUsername) async {
+    username = theUsername;
+  }
+
+  String getPassword()  {
+    return password;
+  }
+
+  static void setPassword(String thePassword) async {
+    password = thePassword;
+  }
+
+  Future<dynamic> userAuth() async {
+
+    String token= await fetchToken();
+    var userid = await fetchId();
```

Comment:
Good job in handling Basic Authentication mechanisms. You made it work and the received token is being used in all subsequent requests

However, the token is being requested multiple times (actually with every request), meaning you are uneccessarily logging in the user multiple times. You already got the token when you called fetchToken() and you saved it into a variable named accessToken. So just use this accessToken with all other requests. fetchToken() should only be called once and never again until the user logs out. I have added print statements in all fetch functions to see the behavior and got this when logging in: (5 requests were sent to backend while only 2 requests are needed)

￼
![flutter fetching token](https://user-images.githubusercontent.com/14883914/197063418-cdb21376-300d-4406-ad27-f43d018d1fc2.png)

Note that fetchToken() and fetchID call the same API (login api) but the first return the token and the latter returns user ID. One request is enough to read both values and assign them to variables (which is already done by you)